.ai/auto-review/github_reviewer.py

```diff
@@ -92,11 +93,31 @@ def main():
             for response in responses:
                 if response.line:
                     result = post_line_comment(github=github, file=file, text=response.text, line=response.line)
+                if not result:
+                    diff_hunks = extract_diff_hunks(file_diffs)
+                    result = post_line_comment_with_diff_hunks(github=github, file=file, text=response.text, line=response.line, diff_hunk=diff_hunks)
                 if not result:
                     result = post_general_comment(github=github, file=file, text=response.text)
                 if not result:
                     raise RepositoryError("Failed to post any comments.")
 
+def extract_diff_hunks(diff_text, context_lines=3):
+    hunks = []
+    # Match each hunk starting with @@ and capture the line and surrounding context
+    hunk_pattern = re.compile(r'(@@ .+ @@)([\s\S]*?)(?=(^@@|^diff|\Z))', re.MULTILINE)
+
+    for match in hunk_pattern.finditer(diff_text):
```

Comment:
109 : Incorrect slice logic `hunk_body[:context_lines + 1]` the correct way to get the required context `hunk_body[max(0, line_number-context_lines):min(len(hunk_body),line_number+context_lines)]` where `line_number` is the target line within the diff. Currently, it just adds the first 3 lines of context and it's not how it should work based on the name `context_lines`.